modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js

```diff
@@ -373,8 +365,17 @@ const Main = ({
 		}
 	};
 
-	const handleSelectButtonClicked = ({portletNamespace}) => {
+	const handleSelectButtonClicked = ({portletNamespace}, event) => {
+		onFocus(event);
+
+		const startFocusDate = new Date();
+
 		Liferay.Util.openSelectionModal({
+			onClose: () =>
+				onBlur({
+					...event,
+					focusDuration: new Date() - startFocusDate,
+				}),
```

Comment:
@marcosapmf, from what I could see from the existing code, the focus duration control is already being done [here](https://github.com/liferay/liferay-portal/blob/93aa771e575d4997752d00ed378af621d7265dd8/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/Field/Field.es.js#L153-L163), thereby if you just invoke `onFocus` and `onBlur` where you need it should be enough. Can you please give it a try?

Reply:
@natocesarrego, you are right! You see, I am still learning things with you :smile: